File: ucare_backend/src/main/java/com/douzone/ucare/service/MedicineService.java
package com.douzone.ucare.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.douzone.ucare.repository.MedicineRepository;
import com.douzone.ucare.vo.MedicineVo;

@Service
public class MedicineService {
	@Autowired
	private MedicineRepository medicineRepository;
	
	public int create(MedicineVo data) {
		return medicineRepository.create(data);
	}
	
	public int excelCreate(MedicineVo data) {
		return medicineRepository.excelCreate(data);
	}

	public List<MedicineVo> retrieveAll() {
		return medicineRepository.retrieveAll();
	}

	public int update(MedicineVo data) {
		return medicineRepository.update(data);
	}

	public int delete(int medicineNo) {
		return medicineRepository.delete(medicineNo);
	}

}

File: ucare_frontend/src/service/medicineService.js
import axios from 'axios';
import { MEDICINE_API_BASE_URL } from './urlConfig'

class medicineService {
  create(data){
    return axios.post(MEDICINE_API_BASE_URL + '/create', data);
  }
  excelCreate(data){
    return axios.post(MEDICINE_API_BASE_URL + '/excelCreate', data);
  }
  retrieveAll(){
    return axios.get(MEDICINE_API_BASE_URL + '/retrieveAll');
  }
  update(data){
    return axios.put(MEDICINE_API_BASE_URL + '/update', data)
  }
  delete(medicineNo){
    return axios.delete(MEDICINE_API_BASE_URL + '/delete/' + medicineNo);
  }
}

export default new medicineService();